usagov/usagov-2021
# USAGOV-1348 docker config for cypress

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1348

## Description

* Added configuration for a docker container for Cypress tests. This is in the `.docker` directory and `docker-compose.yml`. 
* Updated `README.md` with instructions for setting up the container. (Covers Mac only.) 
* Updated `bin/init` to add a template env.local.cypress file
* Added `bin/cypress-ssh`, a convenience script for connecting to the cypress container
* Added `automated_tests` directory with: 
   * `cypress_build`: artifacts for the cypress docker container
   * `e2e-cypress`: all the cypress test scripts. These were on a different branch; the configuration and many of the tests have been updated here. 
* Updated `.gitignore`

## Type of Changes

  - Add screenshot showing what it should look like
- [x] Other - new local configuration with additional docker container

## Testing Instructions



### Requires New Config

### Requires New Content

### Validation Steps
1. Run `docker system prune --all` and `bin/init` as you would to "start from scratch." You do not need to load a fresh database snapshot unless you want to. 

2. Follow the instructions in `README.md` for **Automated tests setup (cypress)**. Check in with me if you get stuck; hopefully we just need to clarify something in the instructions. (If you're running Windows, we should probably collaborate on adding instructions for an X window server on Windows!) 

You should be able to run the tests, both headless and interactive. They are _not_ expected to all pass, but some of them should pass. 

3. If you run the tests (or a subset of them) headless and don't cancel, an HTML report will be written to automated_tests/e2e-cypress/cypress/reports/html/index.html. You should be able to open that file in your web browser and see test results with screenshots where there were failures.

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
4. Build and deploy the changes.
5. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
